src/Components/Layout.tsx

```diff
@@ -1,596 +1,344 @@
+  const [boxClass, setBoxClass] = useState<string>("box-closed");
+  const toggleDropdown = (num: number): void => {
+    if (num != 1 && bodyConfig.tabSelected != num) {
+      if (boxClass == "box-closed" || boxClass == "box-tiny") {
+        setBoxClass("box");
+        setMenuClass("menu-closed");
+      }
+      setBodyConfig({ ...bodyConfig, tabSelected: num });
+    } else if (bodyConfig.tabSelected == num) {
+      if (boxClass == "box-closed") {
+        setBoxClass("box");
+        setMenuClass("menu-closed");
+      } else {
+        setBoxClass("box-closed");
+      }
+    }
+    if (num == 1) {
+      if (menuClass == "menu-closed") {
+        setMenuClass("menu");
+        setBoxClass("box-closed");
+        setBodyConfig({ ...bodyConfig, tabSelected: num });
+      } else {
+        setMenuClass("menu-closed");
+      }
+    }
+  };
+
+  // COMPONENT OBJECTS
+  function getMenuObj() {
+    return {
+      menuError,
+      setMenuError,
+      timerRunning,
+      setTimerRunning,
+      refreshData,
+      setRefreshData,
+      anchorEl,
+      setAnchorEl,
+      menuClass,
+    };
+  }
+
+  function getBodyObj(index: number) {
+    return {
+      applyMasterChanges: applyChanges,
+      boxClass,
+      setBoxClass,
+      index,
+      bodyConfig,
+      setBodyConfig,
+      dateRangeUI,
+      dateRange,
+      dropdownData,
+      setDropdownData,
+      setDateRangeUI,
+      setDateRange: resetDateRange,
+    };
+  }
+
+  function getQuickSelectObj() {
+    return {
+      dateRange,
+      applyChanges,
+      dateRangeUI,
+      boxClass,
+      setDropdownData,
+      dropdownData,
+      recentlySelected,
+      setRecentlySelected,
+      setDateRangeUI,
+    };
+  }
+
+  // REFRESH FUNCTIONS
+  function resetDateRange(previous: DateRange): void {
+    setDateRangeUI(previous.dateRangeUI);
+    previous.setUI();
+    let newObject = new DateRange(previous.localeObj, previous.dateRangeUI);
+    newObject.load(previous);
+    setDateRange(newObject);
+  }
+
+  function applyChanges(dr: DateRange): void {
+    dr.applyChanges();
+    resetDateRange(dr);
+    props.onDateEvent(dr.dates);
+    setStoredRange(dr);
+
+    if (timerRunning) {
+      setTimerRunning(false);
+    }
+    setBoxClass("box-closed");
+  }
+
+  function refreshTime(): void {
+    dateRange.refreshDates();
+    resetDateRange(dateRange);
+
+    props.onDateEvent(dateRange.dates);
+    setTimerRunning(false);
+    setTimerRunning(true);
+  }
+
+  // RENDERING
+  function getApplyText(): JSX.Element {
+    if (timerRunning) {
+      return (
+        <Box className={defaults.classes.flexRow}>
+          <RefreshIcon />
+          <Box ml={1} />
+          Refresh
+        </Box>
+      );
+    } else {
+      return (
+        <Box className={defaults.classes.flexRow}>
+          <KeyboardTabIcon />
+          <Box ml={1} />
+          Update
+        </Box>
+      );
+    }
+  }
+
+  return (
+    <MuiThemeProvider theme={defaults.theme}>
```

Comment:
Another good opportunity to decompose a bunch of these elements into separate files and separate Components.

Reply:
Done